src/uu/wc/src/wc.rs

```diff
@@ -41,17 +42,17 @@ use crate::{
 /// The minimum character width for formatting counts when reading from stdin.
 const MINIMUM_WIDTH: usize = 7;
 
-struct Settings {
+struct Settings<'a> {
     show_bytes: bool,
     show_chars: bool,
     show_lines: bool,
     show_words: bool,
     show_max_line_length: bool,
-    files0_from_path: Option<PathBuf>,
+    files0_from: Option<Input<'a>>,
```

Comment:
If you make things more complicated for performance reasons (which I assume the lifetimes are for), you do kind of need to show that performance improves. I bet it doesn't really matter, because it happens just once and this is not a hot path. The same applies to some of the `Cow` strings. If you just get rid of cloning and do not introduce complexity it's always fine of course.